sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala

```diff
@@ -170,6 +160,31 @@ class JsonFunctionsSuite extends QueryTest with SharedSQLContext {
       Row(Row(1, "haa")) :: Nil)
   }
 
+  test("SPARK-22228: from_json should support also arrays of primitive types") {
+    val dfInt = Seq("[1]", "[2, 3]").toDS()
```

Comment:
Add a `[]` in to the data?

Reply:
This case is tested few lines after. I preferred to treat each specific case separately so that an error points out very easily which case is not handled properly.